pyenttec/__init__.py

```diff
@@ -235,11 +254,20 @@ def close(self):
 
 
 # --- Error handling ---
-class EnttecPortOpenError(Exception):
+class EnttecError(Exception):
+    pass
+
+class EnttecPortOpenError(EnttecError):
+    pass
+
+class EnttecConfigError(EnttecError):
+    pass
+
+class DMXError(Exception):
     pass
 
-class EnttecConfigError(Exception):
+class DMXAddressError(DMXError):
```

Comment:
I'm not sure that we need this complex exception hierarchy (my opinions here have shifted since writing this library).  `DMXAddressError` is basically just `IndexError`, `DMXOverflowError` is `ValueError`.  Does your use case actually use these exception types in normal operation?

Reply:
I think I agree this is unnecessary. But I'm not sure I agree about `ValueError`, I'd be more inclined to use `OverflowError`. It's what I'm catching to replace with DMXOverflowError anyway.

Reply:
`ValueError` seems more appropriate here; the fact that the mechanism by which this value is out of range is overflow is an implementation detail.  From the docs:

`ValueError`: "Raised when a built-in operation or function receives an argument that has the right type but an inappropriate value, and the situation is not described by a more precise exception such as IndexError."

`OverflowError`: "Raised when the result of an arithmetic operation is too large to be represented. This cannot occur for integers (which would rather raise MemoryError than give up). However, for historical reasons, OverflowError is sometimes raised for integers that are outside a required range. Because of the lack of standardization of floating point exception handling in C, most floating point operations are not checked."

I've never encountered `OverflowError` even once in regular Python programming and I'd be surprised if setting a value failed with it.  On the other hand, I'd expect `ValueError` for a value outside of the expected range.

Reply:
Point taken, I'll stick with ValueError